nncf/quantization/algorithms/weight_compression/mixed_precision.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from typing import Dict, List, Optional
+
+import numpy as np
+from numpy import linalg
+
+from nncf.common.logging.track_progress import track
+from nncf.common.utils.registry import Registry
+from nncf.openvino.graph.node_utils import get_const_value
+from nncf.parameters import SensitivityMetric
+from nncf.quantization.algorithms.weight_compression.compression_info import WeightCompressionConfig
+from nncf.quantization.algorithms.weight_compression.compression_info import WeightNodeParams
+from nncf.quantization.algorithms.weight_compression.quantize import _do_integer_quantization
+from nncf.quantization.algorithms.weight_compression.quantize import _get_integer_quantization_error
+
+MIXED_PRECISION_CRITERIA = Registry("mixed_precision_criteria")
+
+
+class MixedPrecisionCriterion:
+    """
+    Assigns mixed quantization scheme (e.g. uniform int8 or non-uniform nf4) for weights based on some criteria.
+
+    """
+
+    def __init__(
+        self,
+        weight_params: List[WeightNodeParams],
+        primary_config: WeightCompressionConfig,
+        ratio: float,
+        activations: Optional[Dict[str, np.ndarray]] = None,
+    ):
+        """
+        :param weight_params: List of information about internal weight nodes. Only internal nodes are considered
+            for mixed precision. The quantization scheme is added to this info.
```

Comment:
I missed what are the internal weight node, where I can find definition for them?